File: sort-colors/sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        i = 0
        numTwos = 0
        while i < len(nums):
            if nums[i] == 0:
                nums.pop(i)
                nums.insert(0, 0)
            elif nums[i] == 2:
                nums.pop(i)
                numTwos += 1
                i -= 1
            i += 1
        for i in range(numTwos):
            nums.append(2)
        return nums

File: sort-colors/sort-colors.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var sortColors = function(nums) {
    let i = 0;
    let numTwos = 0;
    while (i < nums.length) {
        if (nums[i] == 0) {
            nums.splice(i, 1);
            nums.unshift(0);
        }
        else if (nums[i] == 2) {
            nums.splice(i, 1);
            numTwos++;
            i--;
        }
        i++;
    }
    for (let i=0; i<numTwos; i++) {
        nums.push(2);
    }
};
